torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -790,183 +827,12 @@ def local_shards(self) -> List[Shard]:
         """
         return self._local_shards
 
-    def size(self, dim: int = None) -> Union[torch.Size, int]:
-        """
-        Returns a :Union:`[torch.Size, int]` which represents the size of the tensor.
-            The dimension can be specified.
-
-        Args:
-            dim (int, optional): the dimension over which the size represents.
-                If specified, it returns the size of the given dimension.
-                If not, it returns a subclass of tuple.
-                Default: ``None``
-
-        Returns:
-            A :Union:`[torch.Size, int]` represents the size of the tensor.
-        """
-        size = self._metadata.size
-        if dim is None:
-            return size
-        if dim < -len(size) or dim >= len(size):
-            raise ValueError(
-                "Argument ``dim`` must be within the range of tensor "
-                f"dimensions [-{len(size)}, {len(size)})"
-            )
-        return size[dim]
-
-
-    def is_pinned(self) -> bool:
+    def is_pinned(self) -> bool:  # type: ignore[override]
```

Comment:
Can't this be registered as a default tensor op too?

Reply:
So the issue with `is_pinned` is that it's actually getting dispatch to C++ and then to `torch_dispatch`, if we want to register it the way as default op, we will need to change `torch_dispatch` to handle this call in a special manner. I can think of how to do this in a follow up PR.